endpoints/setuid_test.go

```diff
@@ -18,67 +19,163 @@ import (
+		},
+		{
+			uri:                   "/setuid?bidder=pubmatic&uid=123",
+			existingSyncs:         map[string]string{"rubicon": "def"},
+			gdprAllowsHostCookies: true,
+			expectedSyncs:         map[string]string{"pubmatic": "123", "rubicon": "def"},
+			expectedResponseCode:  http.StatusOK,
+			description:           "Add the uid for the requested bidder to the list of existing syncs",
+		},
+		{
+			uri:                  "/setuid?bidder=pubmatic&uid=123&gdpr=0",
+			existingSyncs:        nil,
+			expectedSyncs:        map[string]string{"pubmatic": "123"},
+			expectedResponseCode: http.StatusOK,
+			description:          "Don't care about GDPR consent if GDPR is set to 0",
+		},
+		{
+			uri:                  "/setuid?bidder=pubmatic&uid=123",
+			existingSyncs:        nil,
+			expectedSyncs:        nil,
+			expectedResponseCode: http.StatusOK,
+			expectedRespMessage:  "The gdpr_consent string prevents cookies from being saved",
+			description:          "Return err message if the GDPR consent doesn't allow syncs for the given bidder",
+		},
+		{
+			uri:                   "/setuid?uid=123",
+			existingSyncs:         nil,
+			expectedSyncs:         nil,
+			gdprAllowsHostCookies: true,
+			expectedResponseCode:  http.StatusBadRequest,
+			expectedRespMessage:   `"bidder" query param is required`,
+			description:           "Return an error if the bidder param is missing from the request",
+		},
+		{
+			uri:                   "/setuid?bidder=appnexus&uid=123&gdpr=2",
+			existingSyncs:         nil,
+			expectedSyncs:         nil,
+			gdprAllowsHostCookies: true,
+			expectedResponseCode:  http.StatusBadRequest,
+			expectedRespMessage:   "the gdpr query param must be either 0 or 1. You gave 2",
+			description:           "Return an error if GDPR is set to anything else other that 0 or 1",
+		},
+		{
+			uri:                   "/setuid?bidder=appnexus&uid=123&gdpr=1",
+			existingSyncs:         nil,
+			expectedSyncs:         nil,
+			gdprAllowsHostCookies: true,
+			expectedResponseCode:  http.StatusBadRequest,
+			expectedRespMessage:   "gdpr_consent is required when gdpr=1",
+			description:           "Return an error if GDPR is set to 1 but GDPR consent string is missing",
+		},
+		{
+			uri: "/setuid?bidder=pubmatic&uid=123&gdpr_consent=" +
+				"BONciguONcjGKADACHENAOLS1rAHDAFAAEAASABQAMwAeACEAFw",
+			existingSyncs:        nil,
+			expectedSyncs:        nil,
+			gdprReturnsError:     true,
+			expectedResponseCode: http.StatusBadRequest,
+			expectedRespMessage: "No global vendor list was available to interpret this consent string. " +
+				"If this is a new, valid version, it should become available soon.",
+			description: "Return an error if the GDPR string is either malformed or using a newer version that isn't yet supported",
+		},
+		{
+			uri: "/setuid?bidder=pubmatic&uid=123&gdpr=1&gdpr_consent" +
+				"=BONciguONcjGKADACHENAOLS1rAHDAFAAEAASABQAMwAeACEAFw",
```

Comment:
Nitpick: The `=` is on the line with the consent string in this test, but on the above test it's on the previous line. For a moment, I thought the content of the consent string was different due to the inconsistency.

Reply:
Ah! Thanks for pointing that out